adapters/missena/missena.go

```diff
@@ -0,0 +1,229 @@
+package missena
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type MissenaAdRequest struct {
+	RequestId        string `json:"request_id"`
+	Timeout          int    `json:"timeout"`
+	Referer          string `json:"referer"`
+	RefererCanonical string `json:"referer_canonical"`
+	GDPRConsent      string `json:"consent_string"`
+	GDPR             bool   `json:"consent_required"`
+	Placement        string `json:"placement"`
+	TestMode         string `json:"test"`
+}
+
+type MissenaBidServerResponse struct {
+	Ad        string  `json:"ad"`
+	Cpm       float64 `json:"cpm"`
+	Currency  string  `json:"currency"`
+	RequestId string  `json:"requestId"`
+}
+
+type MissenaInternalParams struct {
+	ApiKey           string
+	RequestId        string
+	Timeout          int
+	Referer          string
+	RefererCanonical string
+	GDPRConsent      string
+	GDPR             bool
+	Placement        string
+	TestMode         string
+}
+
+type MissenaAdapter struct {
+	EndpointTemplate *template.Template
+}
+
+// Builder builds a new instance of the Foo adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) makeParameter(missenaParams MissenaInternalParams, request *openrtb2.BidRequest) *MissenaAdRequest {
+	missenaRequest := MissenaAdRequest{
+		RequestId:        request.ID,
+		Timeout:          2000,
+		Referer:          request.Site.Page,
+		RefererCanonical: request.Site.Domain,
+		GDPRConsent:      missenaParams.GDPRConsent,
+		GDPR:             missenaParams.GDPR,
+		Placement:        missenaParams.Placement,
+		TestMode:         missenaParams.TestMode,
+	}
+	return &missenaRequest
+}
+
+func (a *adapter) makeRequest(missenaParams MissenaInternalParams, reqInfo *adapters.ExtraRequestInfo, imp *openrtb2.Imp, request *openrtb2.BidRequest) (*adapters.RequestData, error) {
+
+	url := a.endpoint + "?t=" + missenaParams.ApiKey
+	parameter := a.makeParameter(missenaParams, request)
```

Comment:
You could simplify your adapter a bit eliminating a function call by deleting the `makeParameter` function and moving that function's contents in line so that you construct the `MissenaAdRequest` instance here.